fdbclient/include/fdbclient/BlobGranuleCommon.h

```diff
@@ -205,38 +206,47 @@ struct BlobFilePointerRef {
 
 	BlobFilePointerRef() {}
 
-	BlobFilePointerRef(Arena& to, const std::string& filename, int64_t offset, int64_t length, int64_t fullFileLength)
-	  : filename(to, filename), offset(offset), length(length), fullFileLength(fullFileLength) {}
+	BlobFilePointerRef(Arena& to,
+	                   const std::string& filename,
+	                   int64_t offset,
+	                   int64_t length,
+	                   int64_t fullFileLength,
+	                   Version fileVersion)
+	  : filename(to, filename), offset(offset), length(length), fullFileLength(fullFileLength),
+	    fileVersion(fileVersion) {}
 
 	BlobFilePointerRef(Arena& to,
 	                   const std::string& filename,
 	                   int64_t offset,
 	                   int64_t length,
 	                   int64_t fullFileLength,
+	                   Version fileVersion,
 	                   Optional<BlobGranuleCipherKeysCtx> ciphKeysCtx)
 	  : filename(to, filename), offset(offset), length(length), fullFileLength(fullFileLength),
-	    cipherKeysCtx(ciphKeysCtx) {}
+	    fileVersion(fileVersion), cipherKeysCtx(ciphKeysCtx) {}
 
 	BlobFilePointerRef(Arena& to,
 	                   const std::string& filename,
 	                   int64_t offset,
 	                   int64_t length,
 	                   int64_t fullFileLength,
+	                   Version fileVersion,
 	                   Optional<BlobGranuleCipherKeysMeta> ciphKeysMeta)
-	  : filename(to, filename), offset(offset), length(length), fullFileLength(fullFileLength) {
+	  : filename(to, filename), offset(offset), length(length), fullFileLength(fullFileLength),
+	    fileVersion(fileVersion) {
 		if (ciphKeysMeta.present()) {
 			cipherKeysMetaRef = BlobGranuleCipherKeysMetaRef(to, ciphKeysMeta.get());
 		}
 	}
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		serializer(ar, filename, offset, length, fullFileLength, cipherKeysCtx);
+		serializer(ar, filename, offset, length, fullFileLength, fileVersion, cipherKeysCtx);
```

Comment:
will it impact backward compatibility? e.g if we upgrade from previous version,  can we read old format?

Reply:
This is not serialized in the db, only over the network, so it shouldn't.
Processes will only talk over the network with compatible protocol versions